File: Python/01-Alan's-Number-Game.py
def reverse(n): # function to reverse the number.
    temp,rev,rem = n, 0, 0
    while temp != 0:
        rem = temp % 10
        if temp < 0:
            rem = rem - 10
        rev = rev*10 + rem
        temp = int(temp/10)
    return rev
    
n = int(input()) # taking input.
print(n+reverse(n)) # printing the addition of the number and its reverse.

File: C++/01-Alan's-Number-Game.cpp
#include <iostream>
using namespace std;
#include <stdio.h>
int main(){
    long long int n,temp,rev=0,rem=0;
    cin>>n; //taking input
    temp = n; //Storing number in temporary variable 
    while(temp!=0){ //Reversing the Number
        rem = temp%10;
        rev = rev * 10 + rem;
        temp /= 10;
    }
    cout<<n+rev; //Printing output by adding original number with its reverse
    return 0;
}
